[web components]: Invalid exports cause error when bundling: Module not found: Error: Expecting folder to folder mapping. "./es/*" should end with "/"

### Description

The `exports` key in `package.json` lists some subpath exports that are invalid, specifically [these ones](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/package.json#L20-L21):

```json
"./es/": "./es/*",
"./lib/": "./lib/*",
```

This causes the following error for an import like this:

```js
import CTAMixin, { icons as ctaIcons } from '@carbon/ibmdotcom-web-components/es/component-mixins/cta/cta.js';
```

> ERROR in ./src/index.js 1:0-87
Module not found: Error: Expecting folder to folder mapping. "./es/*" should end with "/"

Per the [node docs](https://nodejs.org/api/packages.html#package-entry-points), this should be resolved by adding a trailing asterisk to the key of each of the above subpath exports.

### Component(s) impacted

Modules available via entry points that fall outside of `./es/components/*"`, for instance `./es/component-mixins/*`.

### Browser

Chrome

### Carbon for IBM.com version

v2.10.0

### Severity

Severity 1 = The design is broken in a critical way that blocks users from completing tasks or damages the brand. Affects major functionality, no workaround.

### Application/website

ibm.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://codesandbox.io/p/github/m4olivei/carbon-exports-bug-repro/main?import=true

### Steps to reproduce the issue (if applicable)

In the codesandbox, run `yarn install && yarn build`

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

--- a/packages/web-components/package.json
+++ b/packages/web-components/package.json
@@ -17,8 +17,8 @@
       "node": "./lib/globals/*",
       "default": "./es/globals/*"
     },
-    "./es/": "./es/*",
-    "./lib/": "./lib/*",
+    "./es/*": "./es/*",
+    "./lib/*": "./lib/*",
     "./custom-elements.json": "./custom-elements.json",
     "./package.json": "./package.json"
   },

